flow/project.py

```diff
@@ -1565,6 +1576,23 @@ def select(operation):
                     del messages[:]     # clear
             if not operations:
                 break   # No more pending operations or execution limits reached.
+
+            # Optionally re-order operations for execution if order argument is provided:
+            if callable(order):
+                operations = list(sorted(operations, key=order))
+            elif order == 'round-robin':
+                groups = [list(group)
+                          for _, group in groupby(operations, key=lambda op: op.job)]
+                operations = list(roundrobin(*groups))
+            elif order == 'random':
+                random.shuffle(operations)
+            elif order is None or order in ('none', 'by-job'):
```

Comment:
Why allow both `None` and `'none'`? Seems unnecessary and increases the API scope.

Reply:
The reason for this is so that we can easily specify 'None' from the command line, consider the following example:
```#!bash
for order in none round-robin random; do
    ./project.py run --order=$order -o foo bar
done
```
If `none` was not a valid argument, one would need to drop the `--order` argument altogether, which means you need to either make the code more verbose:
```#!bash
for order in --order=none --order=round-robin --order=random; do
    ./project.py run $order -o foo bar
done
```
or add an awkward special case with potential code duplication:
```#!bash
./project.py run foo bar
for order in round-robin random; do
    ./project.py run --order=$order -o foo bar
done
```
I've seen this kind of pattern before and would argue that that little bit of redundancy is warranted. If it helps, we can make `none` the documented default argument and avoid the redundancy as part of the API documentation.

Reply:
Good reasoning, I am fine with it this way now that I understand.